app/src/main/java/org/openobservatory/ooniprobe/fragment/ProgressFragment.java

```diff
@@ -123,53 +125,44 @@ public void onDestroy() {
         LocalBroadcastManager.getInstance(getActivity()).unregisterReceiver(receiver);
     }
 
-    @Override
-    public void onServiceConnected(ComponentName cname, IBinder binder) {
-        //Bind the service to this activity
-        RunTestService.TestBinder b = (RunTestService.TestBinder) binder;
-        service = b.getService();
-        isBound = true;
-        if (((Application)getActivity().getApplication()).isTestRunning())
-            updateUI();
-    }
+    private class TestRunnerEventListener implements TestRunBroadRequestReceiver.EventListener {
+        @Override
+        public void onStart(RunTestService service) {
+            updateUI(service);
+        }
 
-    @Override
-    public void onServiceDisconnected(ComponentName name) {
-        service = null;
-    }
+        @Override
+        public void onRun(String value) {
+            name.setText(value);
+        }
 
-    public class TestRunBroadRequestReceiver extends BroadcastReceiver {
         @Override
-        public void onReceive(Context context, Intent intent) {
-            String key = intent.getStringExtra("key");
-            String value = intent.getStringExtra("value");
-            switch (key) {
-                case TestAsyncTask.START:
-                    progress.setIndeterminate(true);
-                    if (service != null && service.task != null && service.task.currentSuite != null)
-                        progress.setMax(service.task.currentSuite.getTestList(preferenceManager).length * 100);
-                    break;
-                case TestAsyncTask.RUN:
-                    name.setText(value);
-                    break;
-                case TestAsyncTask.PRG:
-                    if (progress.isIndeterminate())
-                        if (service != null && service.task != null && service.task.currentSuite != null)
-                            progress.setMax(service.task.currentSuite.getTestList(preferenceManager).length * 100);
-                    int prgs = Integer.parseInt(value);
-                    progress.setIndeterminate(false);
-                    progress.setProgress(prgs);
-                    break;
-                case TestAsyncTask.URL:
-                    progress.setIndeterminate(false);
-                    break;
-                case TestAsyncTask.INT:
-                    running.setText(getString(R.string.Dashboard_Running_Stopping_Title));
-                    break;
-                case TestAsyncTask.END:
-                    progress_layout.setVisibility(View.GONE);
-                    break;
-            }
+        public void onProgress(int state, double eta) {
+            if (progress.isIndeterminate())
+                updateUI(receiver.service);
+            progress.setIndeterminate(false);
+            progress.setProgress(state);
+        }
+
+        @Override
+        public void onLog(String value) {}
+
+        @Override
+        public void onError(String value) {}
```

Comment:
```suggestion
        public void onError(String value) {
            /* nothing */
        }
```

Let's make it explicit that the callback is empty on purpose!